_posts/2017-10-12-Rust-1.21.md

```diff
@@ -0,0 +1,210 @@
+is a systems programming language focused on safety, speed, and concurrency.
+
+If you have a previous version of Rust installed, getting Rust 1.21 is as easy as:
+
+```bash
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.21.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1210-2017-10-12
+
+### What's in 1.21.0 stable
+
+This release contains some very minor, but nice-to-have features, as well as
+some new documentation.
+
+First up, a small change to literals. Consider code like this:
+
+```rust
+let x = &5;
+```
+
+In Rust, this code is synonymous with:
+
+```rust
+let _x = 5;
+let x = &x;
+```
+
+That is, the `5` here will be stored on the stack, or possibly in registers.
+`x` will be a reference to it.
+
+However, given that it's a literal integer, there's no reason that it *has*
+to be local like this. Imagine we had a function that took a `'static` argument,
+like `std::thread::spawn`. You might use `x` like this:
+
+
+```rust
+use std::thread;
+
+fn main() {
+    let x = &5;
+    
+    thread::spawn(move || {
+        println!("{}", x);
+    }); 
+    
+}
+```
+
+In previous versions of Rust, this would fail to compile:
+
+```text
+error[E0597]: borrowed value does not live long enough
+  --> src/main.rs:4:14
+   |
+4  |     let x = &5;
+   |              ^ does not live long enough
+...
+10 | }
+   | - temporary value only lives until here
+   |
+   = note: borrowed value must be valid for the static lifetime...
+```
+
+Because the `5` is local, so is its borrow, which doesn't satisfy the
+requirements for `spawn`.
+
+However, if you compile this on Rust 1.21, it will work. Why? Well,
+if the thing being referred to is okay to put into a `static`, we could
+instead de-sugar `let x = &5;` like this:
+
+```rust
+static FIVE: i32 = 5;
+
+let x = &FIVE;
+```
+
+Here, since the `FIVE` is `static`, `x` is a `&'static i32`. And so this
+is what Rust will now do in this kind of case. For full details, see [RFC 1414],
+which was accepted in January, but started in December of 2015!
+
+[RFC 1414]: https://github.com/rust-lang/rfcs/blob/master/text/1414-rvalue_static_promotion.md
+
+We [now run LLVM in parallel while generating
+code](https://github.com/rust-lang/rust/pull/43506), which should reduce peak
+memory usage.
+
+The [RLS](https://github.com/rust-lang-nursery/rls/) can now be installed
+[through rustup](https://github.com/rust-lang/rust/pull/44204) by invoking
+`rustup component add rls-preview`. In general, many useful Rust developer
+tools such as the RLS, Clippy, and `rustfmt` need nightly Rust; this is the
+first steps toward having them work on stable Rust. Please check out the
+preview, and you'll hear more about these plans in the future.
```

Comment:
Maybe link to an RLS blog post or something?

Reply:
yeah i wasn't sure what the best place to link was; @nrc any ideas?